Feature/https linux updates (#23379)
dotnet/AspNetCore.Docs: aspnetcore/security/enforcing-ssl.md

See [HTTPS Error using IIS Express (dotnet/AspNetCore #16892)](https://github.com/dotnet/AspNetCore/issues/16892) for troubleshooting certificate issues with Visual Studio.

> ### Linux - certificate not trusted

Reviewer: ```suggestion
### Linux certificate not trusted
```

> Check that the certificate you are configuring for trust is the user HTTPS developer certificate that will be used by the dotnet Kestrel server.

Reviewer: ```suggestion
Check that the certificate being configured for trust is the user HTTPS developer certificate that will be used by the Kestrel server.
```

> You can check the current user default HTTPS developer certificate (used by Kestrel) at the following location; the filename will be the SHA1 thumbprint. If you clear this file (via `dotnet dev-certs https --clean`) then it will be regenerated when needed, but with a different thumbprint.

Reviewer: ```suggestion
Check the current user default HTTPS developer Kestrel certificate at the following location:
```

```
ls -la ~/.dotnet/corefx/cryptography/x509stores/my
```

> 

Reviewer: ```suggestion

The HTTPS developer Kestrel certificate file is the SHA1 thumbprint. When the file is deleted via `dotnet dev-certs https --clean`, it will be regenerated when needed with a different thumbprint.
```

> Check the thumbprint of the exported certificate (being trusted) matches:

Reviewer: ```suggestion
Check the thumbprint of the exported certificate matches with the following command:
```

```
openssl x509 -noout -fingerprint -sha1 -inform pem -in /usr/local/share/ca-certificates/aspnet/https.crt
```

> If it does not match, you either have an old certificate, or may have exported a developer certificate for the root user, and will need to export the correct certificate again. You can check the root user certificate at:

Reviewer: ```suggestion
If the certificate doesn't match, it could be one of the following:

* An old certificate.
* An exported a developer certificate for the root user. For this case, export the  certificate.

The root user certificate can be checked at:
```
